tensorflow/compiler/mlir/tosa/transforms/legalize_common.cc

```diff
@@ -2448,10 +2446,14 @@ std::optional<Value> convertStridedSliceOp(
   }
 
   // Step 2: reshape the sliced array
-  SmallVector<int64_t> a2_shape(input_rank * 2);
+  SmallVector<int64_t> a2_shape;
   for (int i = 0; i < input_rank; ++i) {
-    a2_shape[i * 2 + 0] = a1_size[i] == -1 ? -1 : a1_size[i] / abs(strides[i]);
-    a2_shape[i * 2 + 1] = abs(strides[i]);
+    int64_t abs_stride_i = abs(strides[i]);
+    a2_shape.push_back(a1_size[i] == -1 ? -1 : a1_size[i] / abs_stride_i);
```

Comment:
Could we check with isDynaic or the like ? (not use -1 directly)

Reply:
turned out the a1_size[i] == -1 is not same as ShapedType::isDynamic(a1_size[i]). the a1_size[i] is explicitly set to -1 in the code above as "TOSA's sentinal value". tests failed when I tried changing it to isDynamic().